### PR TITLE
APIM-5750 OpenSSL is not available any more

### PR DESCRIPTION
**Pull request by guillaumelamirand**
## Issue

https://gravitee.atlassian.net/browse/APIM-5750

## Description

The scope defined for the dependencies included in the bundle was forced to 'compile' instead of 'runtime', which excludes some dependencies, such as Netty native dependencies (transport or tcnative).

Additionally, BoringSSL was misconfigured and not always available depending on the architecture.

This PR attempts to address both issues.

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-imcgmkyzsl.chromatic.com)

### CODE REVIEW

File: gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-healthcheck/pom.xml [hunk 1]
```diff
@@ -109,6 +109,7 @@
             <groupId>io.netty</groupId>
             <artifactId>netty-tcnative-boringssl-static</artifactId>
             <classifier>${os.detected.classifier}</classifier>
```
Comment (phiz71) [inline gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-healthcheck/pom.xml:111]: shouldn't we remove this line also ?

File: gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-container/pom.xml [hunk 2]
```diff
@@ -240,14 +240,6 @@
...
             <scope>compile</scope>
         </dependency>
-
-        <!-- Mapstruct + Lombok -->
-        <dependency>
-            <groupId>org.mapstruct</groupId>
-            <artifactId>mapstruct</artifactId>
-            <scope>compile</scope>
-        </dependency>
-
```
Comment (phiz71) [inline gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-container/pom.xml]: 👍🏻 👍🏻 👍🏻 👍🏻 👍🏻

File: gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-distribution/src/main/assembly/plugin-assembly-distribution.xml [hunk 3]
```diff
@@ -132,8 +131,8 @@
                 <exclude>com.graviteesource.*:*:*</exclude>
                 <exclude>org.projectlombok:*:*</exclude>
                 <exclude>org.mapstruct:mapstruct-processor:*</exclude>
+                <exclude>*:*:*:windows-x86_64</exclude>
             </excludes>
-            <scope>compile</scope>
```
Comment (phiz71) [inline gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-distribution/src/main/assembly/plugin-assembly-distribution.xml:136]: Have you checked the consequence on the content of the lib/ext folder? And the size of the final bundle?

File: gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-healthcheck/pom.xml (hunk 1 above)
Comment (guillaumelamirand) [reply gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-healthcheck/pom.xml:111]: no, it is for test purpose, si must be aligned with the host.

File: gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-distribution/src/main/assembly/plugin-assembly-distribution.xml (hunk 3 above)
Comment (guillaumelamirand) [reply gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-distribution/src/main/assembly/plugin-assembly-distribution.xml:136]: I update the assembly accordingly, so shouldn't have any consequence.

**Review by phiz71 — APPROVED** [review]

**Pull request merged**